koku/koku/dev_middleware.py

```diff
@@ -36,14 +42,31 @@ def process_request(self, request):  # pylint: disable=no-self-use
 
         """
         if hasattr(request, "META"):
-            identity_header = {
-                "identity": {
-                    "account_number": "10001",
-                    "type": "User",
-                    "user": {"username": "user_dev", "email": "user_dev@foo.com", "is_org_admin": True},
-                },
-                "entitlements": {"cost_management": {"is_entitled": True}},
-            }
+            if settings.DEVELOPMENT_IDENTITY:
+                identity_header = settings.DEVELOPMENT_IDENTITY
+
+                user_dict = identity_header.get("identity", {}).get("user")
+                user = Mock(
+                    spec=User,
+                    access=user_dict.get("access"),
+                    username=user_dict.get("username", "user_dev"),
+                    email=user_dict.get("email", "user_dev@foo.com"),
+                    admin=user_dict.get("is_org_admin", False),
+                    customer=Mock(account_id=identity_header.get("account_number", "10001")),
+                    req_id="DEVELOPMENT",
+                )
+
+                request.user = user
+            else:
+                identity_header = {
+                    "identity": {
+                        "account_number": "10001",
+                        "type": "User",
+                        "user": {"username": "user_dev", "email": "user_dev@foo.com", "is_org_admin": True},
+                    },
+                    "entitlements": {"cost_management": {"is_entitled": True}},
+                }
```

Comment:
Just a question here, but do we _not_ want to mock a user for `request.user` here?

Reply:
I'm inserting new behavior while leaving the previous behavior unchanged. My main reasoning is just that I don't know if my chosen approach is a good one. If it is a good idea, I can certainly tighten up the logic.

It feels like a bit of a kludge to me to be using Mock outside of the unit tests. I needed an object that would take a dict as input and expose the dict's keys as class attributes. `unittest.Mock` was a solution that works a little better than `collections.UserDict` and is less work than sub-classing `dict`.

I see Mock as "not the best choice, but an expedient choice to show a working example of the idea."

Reply:
You could always steal `dicta` from here: https://github.com/project-koku/nise/blob/9347f801a5d6799ff8795f76c482704a5f63cc84/utility/generate_static_ocp_settings.py#L50